### PR TITLE
Documentation for the changes in private registry credentials

### PR DESCRIPTION
**Pull request by ifbyol**
This PR adds documentation for the changes we are adding for private registry credentials that will go out in `1.14`. The main changes are:
* Added an `Upgrading to Okteto 1.14.x` section explaining how the migration is done for instances using `privateRegistry` key.
* Removed the helm setting of `privateRegistry` from the settings page. We don't want people using that setting anymore from `1.14`
* Added more information in the registry credentials section of the admin dashboard explaining the new functionality (screenshots are still missing) and the advanced section in which we explain how to create the private registry credentials directly in the Kubernetes cluster

I had doubts about creating a specific page for the private registry, but I didn't find any place where it would fit, so I decided to go with the simplest approach and include it within the sections already exist.

### CODE REVIEW

File: src/content/administration/dashboard.mdx [hunk 1]
```diff
@@ -177,12 +177,64 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
+      key: <field in referenced secret> # key of the secret which contains the access key
+      name: <secret-name> # name of the secret where the access key is stored
+    secretAccessKeySecretRef:
+      key: <field in referenced secret> # key of the secret which contains the access secret key
+      name: <secret-name> # name of the secret where the access secret is stored
+```
 
-<p align="center"><Image src="/docs/administration/images/registry_credentials_detail.png" alt="Registry credentials detail" width="700" /></p>
\ No newline at end of file
+Last thing worth to mention is related to the label `app.kubernetes.io/managed-by: okteto` in the CRDs. Adding that label means that the credentials will be editable from the UI, so any admin could come to this view and edit them. If the lifecycle of those credentials is managed directly by you outside Okteto, you can remove that label. Okteto will keep considering those credentials to be used, and they will keep showing in this view but as read only.
```
Comment (pchico83) [inline src/content/administration/dashboard.mdx:240]: I think this is always the case when using CRDs. I would remove the label `app.kubernetes.io/managed-by: okteto` from the samples and its explanation

File: src/content/administration/dashboard.mdx [hunk 2]
```diff
@@ -177,12 +177,64 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
 <p align="center"><Image src="/docs/administration/images/registry_credentials_view.png" alt="Registry credentials view" width="900" /></p>
 
-For each configured registry you can see the non-sensitive information like username or hostname. For sensitive data you can only see some of the last characters as a hint to verify the value.
+Apart of adding new credentials, this view also allows you to rotate those credentials or delete them if you don't needed them anymore. For sensitive data you only see some of the last characters as a hint to verify the value.
+
+<p align="center"><Image src="/docs/administration/images/registry_credentials_detail.png" alt="Registry credentials detail" width="700" /></p>
+
+> Be aware that any change you do in those credentials might take a few minutes to be propagated to all the components in the cluster.
+
+### Advanced configuration
```
Comment (pchico83) [inline src/content/administration/dashboard.mdx:194]: I would move this section to another doc. Maybe a doc in `Self-hosted > Configuration`? Or in the community forum?

File: src/content/self-hosted/install/upgrade.mdx [hunk 3]
```diff
@@ -25,6 +25,20 @@ $ helm upgrade okteto okteto/okteto -f config.yaml --namespace=okteto --version
...
+
+If you don't use `privateRegistry` key, you can just upgrade Okteto as any other release. If you use it, we recommend you to follow these steps to upgrade your instance to `1.14.x` or higher:
+
+- Make an upgrade to the target version keeping the `privateRegistry` section in your helm configuration.
+- Verify that credentials were successfully migrated. You can do that listing the resources `registries.admin.okteto.com` (`kubectl get registries.admin.okteto.com -n okteto`) present in your cluster. You should see one resource per registry you had configured. Those resources are created by Okteto as part of the upgrade process and they keep the metadata and a reference to the secret which contains the credentials.
```
Comment (pchico83) [inline src/content/self-hosted/install/upgrade.mdx:37]: Why not checking the Admin UI? I think that's easier than checking the kubernetes resources

File: src/content/administration/dashboard.mdx (hunk 2 above)
Comment (pchico83) [reply src/content/administration/dashboard.mdx:194]: Maybe a section about CRDs in general. We should document the git catalog CRD in case someone prefers to define the catalog in code.
But I feel CRDs is too low level to be included in a doc that explains the Admin View

File: src/content/administration/dashboard.mdx (hunk 1 above)
Comment (maroshii) [reply src/content/administration/dashboard.mdx:240]: +1. I don't think we should encourage the use of this label to resources created directly via `kubectl`. This could bring problems. With this label enabled (since it's managed by us) we assume a single secret with "username" and "password" keys and a predefined secret name. If they use this label and use different keys or a different secret name it will work initially but it will be editable through the UI and once edited through the UI it will create a new secret.

To recap, if they want to add the label they need to:

1. Use the same secret for user/password
2. The secret keys **must** be `username` and `password`
3. The secret name must be: `{host}-regcreds`

Comment (ifbyol) [reply src/content/administration/dashboard.mdx:240]: Ok, agree to explain it in in the other way, but I would refer somehow to that label or explain it. If someone wants to migrate their current registries defined here and migrated automatically by Okteto to their git ops approach, they would have to remove that label or it will have problems if change the value directly from the UI

File: src/content/self-hosted/install/upgrade.mdx (hunk 3 above)
Comment (ifbyol) [reply src/content/self-hosted/install/upgrade.mdx:37]: Yeah, when I started to write this, we were planning to show both helm and CRD credentials, but now checking the UI should be enough

File: src/content/administration/dashboard.mdx (hunk 2 above)
Comment (ifbyol) [reply src/content/administration/dashboard.mdx:194]: I added it here because there was not specific section for private registries or CRDs. I feel that one specific for private registry might be a small without too many content. One about CRDs might make sense as we can talk about others

File: src/content/self-hosted/install/upgrade.mdx (hunk 3 above)
Comment (ifbyol) [reply src/content/self-hosted/install/upgrade.mdx]: Done!

File: src/content/administration/dashboard.mdx (hunk 1 above)
Comment (ifbyol) [reply src/content/administration/dashboard.mdx:240]: @maroshii Related to this:

```
To recap, if they want to add the label they need to:

* The secret keys must be username and password
* The secret name must be: {host}-regcreds
```

Aren't we reusing the secret and field names defined in the CRD when updating or managing the credentials?

Comment (maroshii) [reply src/content/administration/dashboard.mdx:240]: Yes we are and the read path should not break but when we create/update we make some assumptions. For eg, we assume they are stored in a single secret and we only read `reg.Spec.Static.UserSecretRef.Name`.

The whole point of that label was to be able to make those assumptions and make change without affecting users. I think we should keep that property and not expose these internals.

Comment (ifbyol) [reply src/content/administration/dashboard.mdx]: Ok, thanks for the explanation. I moved all this to another section. It's own page and removed the reference for the label. Take a look when you have a chance

File: src/content/administration/dashboard.mdx (hunk 2 above)
Comment (ifbyol) [reply src/content/administration/dashboard.mdx]: @pchico83  Moved to its own page. Take a look when you have a chance and let me know what you think

File: src/content/administration/dashboard.mdx [hunk 4]
```diff
@@ -177,12 +177,16 @@ From this page you can follow a simple step-by-step guide to get the major compo
 
 ## Registry Credentials
 
-This section in the admin view allows you to see the registry credentials that have been [configured through the helm chart](/self-hosted/administration/configuration/#privateregistry). You will see a row for each configured registry with the type (Static or AWS) and the registry hostname. The 2 types of credentials supported are:
+This section of the Admin dashboard allows you to manage credentials for private registries that you want to use in your Okteto instance. Those credentials are injected by Okteto in any operation (build, deploy, etc...) done via CLI or the UI, so developers don't need access to them. You only need to configure your credentials here, and all the developers will be able to interact with those registries while developing with Okteto. If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes, so the cluster can pull images from those registries.
```
Comment (omnomagonz) [inline src/content/administration/dashboard.mdx:180]:
```suggestion
This section of the Admin dashboard lets you manage credentials for private registries to use in your Okteto instance. The credentials you specify here are injected by Okteto into any operation (e.g. build, deploy) you perform (UI or CLI) so developers don't need access to them. You only need to configure your credentials here and all developers in your Okteto instance will be able to interact with those registries while developing with Okteto. If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes so the cluster can pull images from those registries.
```

Some minor suggested changes 🙂

File: src/content/administration/dashboard.mdx [hunk 5]
```diff
@@ -177,12 +177,16 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
+
+For each configured registry, you will see a row with the type (Static or AWS) and the registry hostname. The 2 types of credentials supported are:
 - Static: To configure registries with static username and password like Docker Hub.
```
Comment (omnomagonz) [inline src/content/administration/dashboard.mdx:183]:
```suggestion
- Static: Registry configuration using static username and password credentials (e.g. Docker Hub)
```

File: src/content/administration/dashboard.mdx [hunk 6]
```diff
@@ -177,12 +177,16 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
-- AWS: To configure [ECR registries](https://docs.aws.amazon.com/ecr/index.html) using Access Key and Access Secret. In this case, Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured.
+- AWS: To configure [ECR registries](https://docs.aws.amazon.com/ecr/index.html) using Access Key and Access Secret. In this case, Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured. The hostname of the registry has to end with `.amazonaws.com`.
```
Comment (omnomagonz) [inline src/content/administration/dashboard.mdx:184]:
```suggestion
- AWS: [ECR registry](https://docs.aws.amazon.com/ecr/index.html) configuration using Access Key and Access Secret. Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured. The hostname of the registry must end with `.amazonaws.com`.
```

File: src/content/administration/dashboard.mdx [hunk 7]
```diff
@@ -177,12 +177,16 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
 
 <p align="center"><Image src="/docs/administration/images/registry_credentials_view.png" alt="Registry credentials view" width="900" /></p>
 
-For each configured registry you can see the non-sensitive information like username or hostname. For sensitive data you can only see some of the last characters as a hint to verify the value.
+Apart of adding new credentials, this view also allows you to rotate those credentials or delete them if you don't need them anymore. For sensitive data you only see the last few characters as a hint to verify the value.
```
Comment (omnomagonz) [inline src/content/administration/dashboard.mdx:188]:
```suggestion
This admin view also enables you to rotate your registry credentials or delete them if you don't need them anymore. For sensitive data you will only see the last few characters as a hint to verify the value.
```

File: src/content/administration/dashboard.mdx [hunk 8]
```diff
@@ -177,12 +177,16 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
+
+<p align="center"><Image src="/docs/administration/images/registry_credentials_detail.png" alt="Registry credentials detail" width="700" /></p>
 
-<p align="center"><Image src="/docs/administration/images/registry_credentials_detail.png" alt="Registry credentials detail" width="700" /></p>
\ No newline at end of file
+> Be aware that any change you do in those credentials might take a few minutes to be propagated to all the components in the cluster.
```
Comment (omnomagonz) [inline src/content/administration/dashboard.mdx:192]:
```suggestion
> Be aware that any changes you make to credentials may take a few minutes to propagate to all of the components in the cluster.
```

File: src/content/self-hosted/install/crds.mdx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+```console
+kubectl get crd | grep okteto.com
+```
+
+You can let Okteto install these CRDs or you can install them manually on your own, but by aware they are needed for Okteto to work properly. You can use the [following configuration setting](/self-hosted/administration/configuration.mdx/#crds) to let Okteto know if it should install them or not. By default, Okteto will install them.
+
+## Private Registries
+
+As part of the changes for `1.14`, Okteto has changed the way it handles the configuration for private registries. From that version, they can be configured via the [Admin dashboard](administration/dashboard.mdx/#registry-credentials), or they can be defined directly in your cluster through one of the CRDs installed by Okteto: `admin.okteto.com`. This last case is useful, for example, if you have your own mechanism to provision credentials/secrets in your cluster.
```
Comment (omnomagonz) [inline src/content/self-hosted/install/crds.mdx:21]:
```suggestion
In Okteto version `1.14` we've changed the way private registry configuration is managed. In `1.14` onward you must configure private registries via the [Admin dashboard](administration/dashboard.mdx/#registry-credentials) or by defining them directly in your cluster through one of the CRDs installed by Okteto: `admin.okteto.com`. This last case is useful, for example, if you have your own mechanism to provision credentials/secrets in your cluster.
```

**Comment by ifbyol** [comment]
I just set the PR ready for review as I already did the latest changes and updated a couple of screenshots

File: src/content/administration/dashboard.mdx [hunk 10]
```diff
@@ -177,12 +177,16 @@ From this page you can follow a simple step-by-step guide to get the major compo
...
-- Static: To configure registries with static username and password like Docker Hub.
-- AWS: To configure [ECR registries](https://docs.aws.amazon.com/ecr/index.html) using Access Key and Access Secret. In this case, Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured.
+This section of the Admin dashboard lets you manage credentials for private registries to use in your Okteto instance. The credentials you specify here are injected by Okteto into any operation (e.g. build, deploy) you perform (UI or CLI) so developers don't need access to them. You only need to configure your credentials here and all developers in your Okteto instance will be able to interact with those registries while developing with Okteto. If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes so the cluster can pull images from those registries.
```
Comment (omnomagonz) [inline src/content/administration/dashboard.mdx:180]:
```suggestion
In this section of the Admin dashboard you can manage the private registry credentials for your Okteto instance. The credentials you configure here are injected by Okteto into any operation (e.g. build, deploy) performed by developers so they don't need direct access to the credentials. Once you configure the credentials here all developers in your Okteto instance will be able to use the respective registries while developing with Okteto. 

If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes so the cluster can also pull images from those registries.
```

I took a stab at breaking this up a bit to make it a bit easier to digest and also make it a bit more concise

File: src/content/self-hosted/install/crds.mdx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+As part of Okteto's installation, some Custom Resource Definitions (CRDs) are installed in your cluster. These CRDs are used by Okteto to mainly store different information. You can see the list of CRDs installed by Okteto exploring the folder `crds` in the templates of the [Okteto Helm Chart](https://artifacthub.io/packages/helm/okteto/okteto) or by running the following command in your cluster:
+
+```console
+kubectl get crd | grep okteto.com
+```
+
+You can let Okteto install these CRDs or you can install them manually on your own, but by aware they are needed for Okteto to work properly. You can use the [following configuration setting](/self-hosted/administration/configuration.mdx/#crds) to let Okteto know if it should install them or not. By default, Okteto will install them.
```
Comment (omnomagonz) [inline src/content/self-hosted/install/crds.mdx:17]:
```suggestion
You can let Okteto install these CRDs or you can install them manually on your own, but be aware they are needed for Okteto to work properly. You can use the [following configuration setting](/self-hosted/administration/configuration.mdx/#crds) to let Okteto know if it should install them or not. By default, Okteto will install them.
```

File: src/content/self-hosted/install/crds.mdx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+## Private Registries
+
+In Okteto version `1.14` we've changed the way private registry configuration is managed. In `1.14` onward you must configure private registries via the [Admin dashboard](administration/dashboard.mdx/#registry-credentials) or by defining them directly in your cluster through one of the CRDs installed by Okteto: `admin.okteto.com`. This last case is useful, for example, if you have your own mechanism to provision credentials/secrets in your cluster.
+
+If you prefer to define the registries directly in your cluster, you can do it creating an [Opaque Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret) in the cluster with the data needed for the registry. Depending on the type of registry, you will need different fields. Once the secret is created, you will need to create a resource with the apiVersion `admin.okteto.com/v1`, which is one of the CRDs installed by Okteto, refering to the secret which contains the credentials.
```
Comment (omnomagonz) [inline src/content/self-hosted/install/crds.mdx:23]:
```suggestion
If you prefer to define the registries directly in your cluster, you can do it by creating an [Opaque Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret) in the cluster with the data needed for the registry. Depending on the type of registry, you will need different fields. 

Once the secret is created, you will need to create a resource with the apiVersion `admin.okteto.com/v1`, which is one of the CRDs installed by Okteto, referring to the secret which contains the credentials.
```

File: src/content/self-hosted/install/crds.mdx [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+  awsIamUser:
+    accessKeyIDSecretRef:
+      key: <field in referenced secret> # key of the secret which contains the access key
+      name: <secret-name> # name of the secret where the access key is stored
+    secretAccessKeySecretRef:
+      key: <field in referenced secret> # key of the secret which contains the access secret key
+      name: <secret-name> # name of the secret where the access secret is stored
+```
+
+> Be aware that any change you do in those credentials might take a few minutes to be propagated to all the components in the cluster. To make sure Okteto is reading properly those registries, you can check it going to the [Admin dashboard](administration/dashboard.mdx/#registry-credentials). They will be displayed, but they can't be modified from the UI. If you want to manage them from the UI, they have to be created from there.
```
Comment (omnomagonz) [inline src/content/self-hosted/install/crds.mdx:61]:
```suggestion
:::note
Any changes you make to credentials can take a few minutes to propagate to all components in the cluster. To make sure Okteto is able to access the registries, you can check in the [Admin dashboard](administration/dashboard.mdx/#registry-credentials). They will be displayed, but they can't be modified from the UI. If you want to manage them from the UI, they have to be created from there.
:::
```

Is the last sentence still valid with these latest changes? I assume it may be accurate and specific to AWS but wanted to double check.

I may have further suggestions on this copy depending on the clarification here 😄

File: src/content/self-hosted/install/upgrade.mdx [hunk 14]
```diff
@@ -25,6 +25,20 @@ $ helm upgrade okteto okteto/okteto -f config.yaml --namespace=okteto --version
...
 
+## Upgrading to Okteto 1.14.x
+
+As part of [1.14.0](self-hosted/install/releases.mdx#1140), we have changed the way you can configure credentials for private registries. Before, you had to specify them in your helm configuration under `privateRegistry` key. From now on, that section (`privateRegistry`) is deprecated and we have added new options in the [Admin view](administration/dashboard.mdx#registry-credentials) to manage them there.
+
+In order to simplify the migration for all the instances already using that setting, Okteto will automatically migrate the credentials specified in the helm chart to secrets within the cluster **as part of the upgrade process to `1.14.x`**. So, from that moment, any change in those credentials has to be done through the Admin View, and you can remove the `privateRegistry` section from your helm configuration and upgrade Okteto again to completely stop using the deprecated `privateRegistry` section, as they don't apply anymore.
+
+If you don't use `privateRegistry` key, you can just upgrade Okteto as any other release. If you use it, we recommend you to follow these steps to upgrade your instance to `1.14.x` or higher:
```
Comment (omnomagonz) [inline src/content/self-hosted/install/upgrade.mdx:34]:
```suggestion
If you don't use `privateRegistry` key, you can just upgrade Okteto as any other release. If you do use it, we recommend you follow these steps to upgrade your instance to `1.14.x` or higher:
```

File: src/content/self-hosted/install/crds.mdx (hunk 13 above)
Comment (ifbyol) [reply src/content/self-hosted/install/crds.mdx:61]: This note is for both types (static and AWS) and it is still valid. If you create the credentials in this way (using CRDs), you will only see them in the UI but you cannot modify them. If you want to be able to edit them from the UI, you would need to create them from there

Comment (omnomagonz) [reply src/content/self-hosted/install/crds.mdx:63]: Thanks for the clarification 🙇 

I updated my proposed change based on that but let me know if you disagree with it

**Review by omnomagonz — APPROVED** [review]

File: src/content/self-hosted/install/crds.mdx (hunk 13 above)
Comment (ifbyol) [reply src/content/self-hosted/install/crds.mdx]: Your proposal looks good, I just commit it
